controllers/compaction/reconciler.go

```diff
@@ -96,23 +96,78 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		}, err
 	}
 
-	if !etcd.DeletionTimestamp.IsZero() {
+	if !etcd.DeletionTimestamp.IsZero() || etcd.Spec.Backup.Store == nil {
 		// Delete compaction job if exists
 		return r.delete(ctx, r.logger, etcd)
 	}
 
-	if etcd.Spec.Backup.Store == nil {
-		return ctrl.Result{}, nil
+	logger := r.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String())
+
+	return r.reconcileJob(ctx, logger, etcd)
+}
+
+func (r *Reconciler) reconcileJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
+	// Update metrics for currently running compaction job, if any
+	job := &batchv1.Job{}
+	err := r.Get(ctx, types.NamespacedName{Name: etcd.GetCompactionJobName(), Namespace: etcd.Namespace}, job)
+	if err != nil {
```

Comment:
We are treating NotFound error and other errors as same. If there is an error getting a Job should we not re-queue instead of just logging it? In case it is not found then we move ahead.

Reply:
Made the change